pkg/apply/prune/grouping.go

```diff
@@ -76,120 +75,128 @@ func FindGroupingObject(infos []*resource.Info) (*resource.Info, bool) {
 	return nil, false
 }
 
-// SortGroupingObject reorders the infos slice to place the grouping
-// object in the first position. Returns true if grouping object found,
-// false otherwise.
-func SortGroupingObject(infos []*resource.Info) bool {
-	for i, info := range infos {
-		if info != nil && IsGroupingObject(info.Object) {
-			// If the grouping object is not already in the first position,
-			// swap the grouping object with the first object.
-			if i > 0 {
-				infos[0], infos[i] = infos[i], infos[0]
-			}
-			return true
-		}
-	}
-	return false
-}
-
-// PrependGroupingObject orders the objects to apply so the "grouping"
-// object stores the inventory, and it is first to be applied.
-func PrependGroupingObject(o *apply.ApplyOptions) func() error {
-	return func() error {
-		if o == nil {
-			return fmt.Errorf("ApplyOptions are nil")
-		}
-		infos, err := o.GetObjects()
-		if err != nil {
-			return err
-		}
-		_, exists := FindGroupingObject(infos)
-		if !exists {
-			return NoGroupingObjError{}
-		}
-		if err := AddInventoryToGroupingObj(infos); err != nil {
-			return err
-		}
-		if !SortGroupingObject(infos) {
-			return err
-		}
-		return nil
-	}
-}
-
-// Adds the inventory of all objects (passed as infos) to the
-// grouping object. Returns an error if a grouping object does not
-// exist, or we are unable to successfully add the inventory to
-// the grouping object; nil otherwise. Each object is in
-// unstructured.Unstructured format.
-func AddInventoryToGroupingObj(infos []*resource.Info) error {
+// AddInventoryToGroupingObj takes a slice of infos, finds the
+// grouping object template among the resources, and returns a
+// new slice of infos that has a fully populated grouping object
+// as the first info object in the slice. It returns an error
+// if no grouping object can be found, or if the resources
+// do not have the expected unstructured format.
+func AddInventoryToGroupingObj(infos []*resource.Info) ([]*resource.Info,
```

Comment:
I don't see the reason for re-writing this function `AddInventoryToGroupingObj`. If it's a refactor to clean-up, and not a bug fix, then let's move this to another PR.

Reply:
Yeah, I've cleaned this up.